Update fragment format spec (#4416)
TileDB-Inc/TileDB: format_spec/fragment.md

| Variable tile offsets for attribute/dimension 1 | [Tile Offsets](#tile-offsets) | The serialized variable tile offsets for attribute/dimension 1 |
| … | … | … |

> | Variable tile offsets for attribute/dimension N | [Tile Offsets](#tile-offsets) | The serialized variable tile offsets for attribute/dimension N |

Reviewer: Add _on-disk_ for all tile offsets members for consistency.

> | Variable tile sizes for attribute/dimension 1 | [Tile Sizes](#tile-sizes) | The on-disk serialized variable tile sizes for attribute/dimension 1 |

Reviewer: ```suggestion
| Variable tile sizes for attribute/dimension 1 | [Tile Sizes](#tile-sizes) | The in-memory serialized variable tile sizes for attribute/dimension 1 |
```

> | Variable tile sizes for attribute/dimension 1 | [Tile Offsets](#tile-offsets) | The on-disk serialized variable tile sizes for attribute/dimension 1 |

Reviewer: Change [Tile offsets] to [Tile sizes] and the #

| … | … | … |

> | Variable tile sizes for attribute/dimension N | [Tile Sizes](#tile-sizes) | The on-disk serialized variable tile sizes for attribute/dimension N |

Reviewer: ```suggestion
| Variable tile sizes for attribute/dimension N | [Tile Sizes](#tile-sizes) | The in-memory serialized variable tile sizes for attribute/dimension N |
```

### Tile Offsets

> Tile offsets refer to each on-disk data tile's starting byte offset. \

Reviewer: ```suggestion
Tile offsets refer to each _on-disk_ data tile's starting byte offset. \
```

> Tile offsets should be set _only_ on variable-sized tiles, to refer to each data tile's starting byte offset. \

Reviewer: That is incorrect. The tile offsets are for any on disk tiles. They might be compressed so we cannot calculate where they are on disk.

### Tile Sizes

> The tile size refers to the _on-disk_ size. \

Reviewer: Those actually refer to the in memory tile sizes. This information needs to be stored because there is no way to calculate how big a var tile is in memory.
